util/settings/settings.go

```diff
@@ -807,6 +807,34 @@ func (mgr *SettingsManager) GetInstallationID() (string, error) {
 	return argoCDCM.Data[settingsInstallationID], nil
 }
 
+func (mgr *SettingsManager) GetRepositorySecretAccessToken(repoURL string) (string, error) {
+	selector, err := labels.Parse("argocd.argoproj.io/secret-type=repository")
+	if err != nil {
+		return "", fmt.Errorf("failed to parse Argo CD repository selector: %w", err)
+	}
+
+	secretList, err := mgr.secrets.Secrets(mgr.namespace).List(selector)
+	if err != nil {
+		return "", fmt.Errorf("failed to list repository secrets: %w", err)
+	}
+
+	for _, secret := range secretList {
+		secretRepoURLBytes, ok := secret.Data["url"]
+		if !ok {
+			continue
+		}
+
+		secretRepoURL := string(secretRepoURLBytes)
+		if secretRepoURL == repoURL {
+			if tokenBytes, exists := secret.Data["accessToken"]; exists {
+				return string(tokenBytes), nil
+			}
```

Comment:
I'm a bit uncomfortable with using selector and processing multiple items. What if somebody hijacks the flow by adding a new secret with a different/wrong token?

Reply:
Could you please elaborate a bit more about this?

Reply:
Why don't we just query a specific secret?

Reply:
well see, if we are adding a private repo the secret that are created are of the format ```repo-178100904``` , so you cannot directly query a specific secret, you have to iterate all the repository type secret and the select the one that matches with the URL.